### DISCUSSION TITLE
Box64 on GPU?

### DISCUSSION THREAD

**Discussion by Guillermo-Ramirez-Jimenez** [ptitSeb/box64#1934] [Ideas]
Hello,
I am a noob in the CPU world but, I had a crazy idea which you may help me clarify.
When testing some emulators on my phone (SD695), mainly Winlator and Mobox, I noticed the bottleneck is caused by the CPU.
For example, I tested Dark Souls 2 SOTFS and gave me 10-25 fps.
The interesting thing is the performance did not really change with resolution, as I was able to get the same framerate on a resolution of 640x360 and a GPU load of 20-40%.
So, the question is: is it possible to offload the CPU by moving the CPU translation to the GPU using, for example, OpenCL?
We still have the OS emulation on the CPU with Wine but, I think it would help.
Thanks.

**Comment by ptitSeb** [comment]
That would be very tricky to do. and would be effective only if moving entire thread to the GPU.

One thing to note is that, some games are still 32bits in Windows world (it's the case for the original DS2, the remake is 64bits). Depending on the emulator you are running, 32bits might be run with box86 (that does 32bits emulation) or with box64 (64bits emulation) using a special build of Wine "new Wow64" (that does the 64bits <-> 32bits translation). Many modern SnapDragon CPU have Prime core that can only run 64bits code (and so box64).

Also , note that, for the original, the minimum specs is a 3GHz CPU. That's pretty high spec actually.

**Comment by Guillermo-Ramirez-Jimenez** [comment]
I think Box64+Wine WOW64 is the standard for PC game emulation on Android now.
How can we estimate if a game or program will run smoothly then?
Trial and error takes a lot of time with big games because of the installation time.

**Reply by ptitSeb** [reply]
I don't have any miracle solution here. Trial and error (and sharing of the results) is still the best way.
